catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -891,41 +891,31 @@ private Object interpretObjectSensor(Sensors sensor, Sprite sprite) {
 	}
 
 	private Double interpretOperatorEqual(Object left, Object right) {
-
-		if (left instanceof String && right instanceof String) {
-			try {
-				return (Double.valueOf((String) left).compareTo(Double.valueOf((String) right))) == 0 ? 1d : 0;
-			} catch (NumberFormatException numberFormatException) {
-				int compareResult = ((String) left).compareTo((String) right);
-				if (compareResult == 0) {
-					return 1d;
-				}
-			}
-		}
-		if (left instanceof Double && right instanceof String) {
-			try {
-				int compareResult = ((Double) left).compareTo(Double.valueOf((String) right));
-				if (compareResult == 0) {
-					return 1d;
-				}
-			} catch (NumberFormatException numberFormatException) {
-				return 0d;
+		try {
+			Double tempLeft = Double.valueOf(String.valueOf(left));
+			Double tempRight = Double.valueOf(String.valueOf(right));
+			int compareResult = getCompareResult(tempLeft, tempRight);
+			if (compareResult == 0) {
+				return 1d;
 			}
-		}
-		if (left instanceof String && right instanceof Double) {
-			try {
-				int compareResult = Double.valueOf((String) left).compareTo((Double) right);
-				if (compareResult == 0) {
-					return 1d;
-				}
-			} catch (NumberFormatException numberFormatException) {
-				return 0d;
+			return 0d;
+		} catch (NumberFormatException numberFormatException) {
```

Comment:
@cheidenreich, [ParserTestOperators.java#L142](https://github.com/Catrobat/Catroid/blob/f471d0c3abba61eaf03f56938909b7ea6219d5ca/catroid/src/androidTest/java/org/catrobat/catroid/test/formulaeditor/ParserTestOperators.java#L142) covers many cases. Do you've another test in mind?